lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/swig/SwigUtils.scala

```diff
@@ -169,3 +169,79 @@ class IntSwigArray(val array: SWIGTYPE_p_int) extends BaseSwigArray[Int] {
 
   def delete(): Unit = lightgbmlib.delete_intArray(array)
 }
+
+// Wraps long*
+class LongSwigArray(val array: SWIGTYPE_p_long_long) extends BaseSwigArray[Long] {
+  def this(size: Long) = this(lightgbmlib.new_longArray(size))
+
+  def getItem(index: Long): Long = lightgbmlib.longArray_getitem(array, index)
+
+  def setItem(index: Long, item: Long): Unit = lightgbmlib.longArray_setitem(array, index, item)
+
+  def delete(): Unit = lightgbmlib.delete_longArray(array)
+}
+
+// Wraps void**
+class VoidPointerSwigArray(val array: SWIGTYPE_p_p_void, val size: Int) extends BaseSwigArray[SWIGTYPE_p_void] {
+  def this(size: Int) = this(lightgbmlib.new_voidPtrArray(size), size)
+
+  def getItem(index: Long): SWIGTYPE_p_void = lightgbmlib.voidPtrArray_getitem(array, index)
+
+  def setItem(index: Long, item: SWIGTYPE_p_void): Unit = {
+    lightgbmlib.voidPtrArray_setitem(array, index, item) // set native pointer value
+  }
+
+  def delete(): Unit =  {
+    lightgbmlib.delete_voidPtrArray(array)
+  }
+}
```

Comment:
Is there any way to push some of this stuff back into the bindings themselves? This isnt something thats absolutely necessary right now but might be good to think about once we get this working